tasks/pkg-debian.yml

```diff
@@ -2,7 +2,11 @@
 - apt: name=apt-transport-https state=latest
 
 - apt_key: id=C7A7DA52 keyserver=hkp://keyserver.ubuntu.com:80 state=present
+  when: apt_key_url is not defined
 
-- apt_repository: repo='deb http://apt.datadoghq.com/ stable main' state=present update_cache=yes
+- apt_key: url={{ datadog_apt_key_url }} state=present
```

Comment:
Yes, the id will always be the same. It isn't required, but it will save downloading the key if the id is already present in the target keyring, so I'll add it back in.